src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -0,0 +1,601 @@
+
+
+For example, let's consider the following data:
+
+```surql title="Create a new planet, city, and explorer records"
+CREATE planet:unknown_planet;
+CREATE city:el_dorado          SET name = "El Dorado";
+CREATE explorer:drake          SET name = "Drake";
+CREATE explorer:local_guide    SET name = "Local Guide";
+
+RELATE explorer:drake->discovered->planet:unknown_planet;
+RELATE explorer:drake->visited->city:el_dorado;
+RELATE explorer:local_guide->assisted->explorer:drake;
+
+```
+
+```surql title="Retrieve all relationships from Drake"
+SELECT 
+    *,
+    ->? AS actions, 
+    <-? AS interactions, 
+    <->? AS connections
+FROM explorer:drake;
+```
+
+```json title="Response"
+[
+	{
+		actions: [
+			discovered:sh9zbsz5u705cxv6qgoi,
+			visited:hmtttiqqfa4mt9is1a7j
+		],
+		connections: [
+			assisted:1pv8k3p1wpuf0guf5bvm,
+			discovered:sh9zbsz5u705cxv6qgoi,
+			visited:hmtttiqqfa4mt9is1a7j
+		],
+		id: explorer:drake,
+		interactions: [
+			assisted:1pv8k3p1wpuf0guf5bvm
+		],
+		name: 'Drake'
+	}
+]
+```
+
+Explanation:
+
+- `*`: Selects all fields of `explorer:drake`.
+- `->? AS actions`: Retrieves all outgoing relationships from Drake and aliases them as actions.
+- `<-? AS interactions`: Retrieves all incoming relationships to Drake and aliases them as interactions.
+- `<->? AS connections`: Retrieves all relationships connected to Drake, regardless of direction, and aliases them as connections.
+
+
+### Destructuring
+
+When working with nested data, you can destructure objects using the `.` and `{ ... }` idioms.
+
+For example, 
+
+```surql title="Create a new person record"
+CREATE person:1 SET name = 'John', age = 21, obj = { a: 1, b: 2, c: { d: 3, e: 4, f: 5 } };
+```
+
+```json title="Response"
+[
+	{
+		age: 21,
+		id: person:1,
+		name: 'John',
+		obj: {
+			a: 1,
+			b: 2,
+			c: {
+				d: 3,
+				e: 4,
+				f: 5
+			}
+		}
+	}
+]
+```
+
+```surql
+SELECT obj.{ a, c.{ e, f } } FROM person;
+```
+
+```json title="Response"
+[
+	{
+		obj: {
+			a: 1,
+			c: {
+				e: 4,
+				f: 5
+			}
+		}
+	}
+]
+```
+You can also OMIT fields that you don't want to destructure using the `OMIT` clause.
+
+```surql 
+SELECT * OMIT obj.c.{ d, f } FROM person;
+```
+
+
+```json title="Response"
+[
+	{
+		a: 1,
+		c: {
+			e: 4
+		}
+	}
+]
+```
+
+### Optional Parts
+
+<Since v="v2.0.0" />
+
+The `?` operator is used to indicate that a part is optional (it may not exist) it also allows you to safely access nested data without having to check if the nested data exists and exit an idiom path early when the result is NONE.
+
+
+
+```surql
+SELECT person.spouse?.name FROM person;
```

Comment:
I forgot about this one! I think to really make the case for the operator you can show what happens when it isn't used.